tools/patch-modify

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2017 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+import argparse
+import sys
+
+def loadPatch(filename):
+    """ Load the patch from file
+    """
+    text = ""
+    with open(filename, "r") as fp:
+        text = fp.read()
+
+    # make sure this looks like a patch file
+    if not text.startswith("From "):
+        return None
```

Comment:
This is not how patch files  are defined -- in fact, "From " is even rather problematic (it commonly gets quoted in emails). It's how "git am" formats patch files, but only so that they look like emails. I'd suggest checking `"\n+++ "  in text`. I. e. this depends on whether you can/want to always assume that this is dealing with "git am" files, as opposed to quilt or other patches. Also, the format of `git am` might change in the future too.